src/Image.php

```diff
@@ -170,7 +170,11 @@ private function getSvgSize()
         static $zlibSupport;
 
         if (null === $zlibSupport) {
-            $zlibSupport = \in_array('compress.zlib', stream_get_wrappers(), true);
+            $zlibSupport = \in_array('compress.zlib', stream_get_wrappers(), true)
+                && ($reader = new XMLReader)
+                && @$reader->open('compress.zlib://data:text/xml,<x/>') === true
+                && @$reader->read() === true
+                && @$reader->close() === true;
```

Comment:
I still think it is wrong to test the object instantiation.

```php
if (null === $zlibSupport) {
    $zlibSupport = false;

    if (\in_array('compress.zlib', stream_get_wrappers(), true)) {
        $reader = new XMLReader();

        if (true ===  @$reader->open('compress.zlib://data:text/xml,<x/>')) {
            $zlibSupport = true;
        }

        @$reader->close();
    }
}
```

Reply:
Your code doesn’t test the `read()` method. How about this?

```php
if (null === $zlibSupport) {
    $reader = new XMLReader();
    $zlibSupport =
        \in_array('compress.zlib', stream_get_wrappers(), true)
        && true === @$reader->open('compress.zlib://data:text/xml,<x/>')
        && true === @$reader->read()
    ;
}
```

Reply:
The test file is `'compress.zlib://data:text/xml,<x/>'`. How can this be opened but not read?

Reply:
For example `compress.zlib://data:text/xml,` can be opened but not read.

Reply:
Yes, but that is not our test string, is it?

Reply:
No, but if the `compress.zlib` wrapper doesn’t work correctly it could currupt the `<x/>` and this would result in `read()` returning false.

Reply:
Very unlikely but then just keep it as it is.